front/app/utils/currency/FormattedBudget.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import useAppConfiguration from 'hooks/useAppConfiguration';
+import { isNilOrError } from 'utils/helperUtils';
+
+// i18n
+import { injectIntl } from 'utils/cl-intl';
+import { InjectedIntlProps, FormattedNumber } from 'react-intl';
+import messages from './messages';
+
+interface Props {
+  value: number;
+}
+
+const FormattedBudget = ({
```

Comment:
Nice abstraction, it makes a lot of sense and simplifies the rest of the app code! 
It would be really nice (and relatively easy) to add a unit test for it, as it's being widely used throughout the app. Happy to support you on that, if you want to pair program :)

Reply:
So we'll do this when I get back mid August :)